parquet-hadoop/src/test/java/org/apache/parquet/statistics/RandomValueGenerators.java

```diff
@@ -0,0 +1,395 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.statistics;
+
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.PrimitiveType;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.math.BigInteger;
+import java.util.*;
+
+public class RandomValueGenerators {
+  private static final String ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890";
+
+  public static abstract class RandomValueGenerator<TValueType extends Comparable<TValueType>> implements Iterable<TValueType> {
+    public static abstract class ValueIterator<TValueType> implements Iterator<TValueType> {
+      private int count;
+
+      public ValueIterator(int count) {
+        this.count = count;
+      }
+
+      @Override
+      public void remove() {
+        throw new NotImplementedException();
+      }
+
+      @Override
+      public boolean hasNext() {
+        return (count --> 0);
+      }
+    }
+
+    protected final Random random;
+    protected final PrimitiveType.PrimitiveTypeName forType;
+    protected final int bufferLength;
+    protected final int total;
+    protected final TValueType minimum;
+    protected final TValueType maximum;
+
+
+    protected RandomValueGenerator(long seed, PrimitiveType.PrimitiveTypeName forType, int total) {
+      this.random = new Random(seed);
+      this.forType = forType;
+      this.bufferLength = 0;
+      this.total = total;
+      this.minimum = null;
+      this.maximum = null;
+    }
+
+    protected RandomValueGenerator(long seed, PrimitiveType.PrimitiveTypeName forType, int total, int bufferLength) {
+      this.random = new Random(seed);
+      this.forType = forType;
+      this.bufferLength = bufferLength;
+      this.total = total;
+
+      this.minimum = null;
+      this.maximum = null;
+    }
+
+    protected RandomValueGenerator(long seed, PrimitiveType.PrimitiveTypeName forType, int total, TValueType minimum, TValueType maximum) {
+      this.random = new Random(seed);
+      this.forType = forType;
+      this.bufferLength = 0;
+      this.total = total;
+
+      this.minimum = minimum;
+      this.maximum = maximum;
+    }
+
+    public RandomValueGenerator(long seed, PrimitiveType.PrimitiveTypeName forType, int total, TValueType minimum, TValueType maximum, int bufferLength) {
```

Comment:
I don't think total and bufferLength are needed. If you generate the min and max values from the seed, then you don't need to pass those either. Then you could have a single constructor that takes the seed and a type.